iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -1037,16 +1104,19 @@ static LogicalResult getOrAllocateResultBuffers(
   }
   return success();
 }
+
 /// Convenience wrapper around core allocation function for the case where the
 /// alias is the buffer for the result directly.
 static LogicalResult getOrAllocateResultBuffers(
-    OpBuilder &b, Operation *op, ArrayRef<Value> tiedOperands,
-    BlockAndValueMapping &bvm, BufferizationPlan &plan,
-    WorkgroupMemoryAllocationFn allocationFn) {
-  auto aliasingBuffers = llvm::to_vector<4>(llvm::map_range(
-      tiedOperands, [&](Value v) { return bvm.lookupOrNull(v); }));
-  return getOrAllocateResultBuffers(b, op, tiedOperands, aliasingBuffers, bvm,
-                                    plan, allocationFn);
+    OpBuilder &b, Operation *op, BlockAndValueMapping &bvm,
+    BufferizationPlan &plan, WorkgroupMemoryAllocationFn allocationFn) {
+  auto aliasingBuffers = llvm::to_vector<4>(
+      llvm::map_range(op->getResults(), [&](OpResult result) {
+        Value tiedOperand = getTiedOperandForResult(result, plan);
+        return tiedOperand ? bvm.lookupOrNull(tiedOperand) : tiedOperand;
```

Comment:
Do we need the ternary operator? I think `lookupOrNull` method handle this condition?

Reply:
`tiedOperand` could be nullptr. Without this it segfaults.